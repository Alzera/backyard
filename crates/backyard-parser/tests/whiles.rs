use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "while ($i <= 10) {};").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn no_block() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "while (true) $a = 4;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn with_key() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "while (true): endwhile;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
