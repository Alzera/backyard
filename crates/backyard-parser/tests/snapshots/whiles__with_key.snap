---
source: crates/backyard-parser/tests/whiles.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: while
    condition:
      node_type: boolean
      is_true: true
      loc:
        start:
          line: 1
          column: 7
          offset: 7
        end:
          line: 1
          column: 7
          offset: 7
      leadings: ~
      trailings: ~
    body:
      node_type: block
      statements: []
      loc:
        start:
          line: 1
          column: 12
          offset: 12
        end:
          line: 1
          column: 14
          offset: 14
      leadings: ~
      trailings: ~
    is_short: true
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 1
        column: 14
        offset: 14
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 22
    offset: 22
leadings: ~
trailings: ~
